src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -7544,6 +7544,60 @@ public async Task Step(Task<int> t)
             comp.VerifyEmitDiagnostics(expected);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/79415")]
+        public void CompoundAssignment()
+        {
+            var source = """
+                using System;
+                using System.Threading.Tasks;
+
+                interface I1 
+                {
+                    int this[int i] { get; set; }
+                }
+
+                struct S1 : I1
+                {
+                    public int F1;
+
+                    public int this[int i]
+                    {
+                        get
+                        {
+                            Console.Write(F1);
+                            return 0;
+                        }
+                        set
+                        {
+                            Console.Write(F1);
+                        }
+                    }
+                }
+
+                class Program
+                {
+                    static async Task Main()
+                    {
+                        await Test3<S1>();
+                    }
+
+                    static T GetT<T>() where T : I1 => (T)(object)new S1 { F1 = 123 };
+
+                    static async Task Test3<T>() where T : I1
+                    {
+                       GetT<T>()[0] += await Get1Async();
```

Comment:
>GetT<T>()[0] += await Get1Async(); [](http://example.com/codeflow?start=23&length=34)

Consider strengthening the test and observing order of evaluation in this statement. For example, replacing ```0``` with  a function call and printing something distinct from each function and the accessors involved. With the current output we cannot even tell accessors apart.